mlflow/langchain/__init__.py

```diff
@@ -330,12 +335,14 @@ def load_retriever(persist_directory):
         code=code_dir_subpath,
         predict_stream_fn="predict_stream",
         streamable=streamable,
+        model_code=model_code_dir_subpath,
         **model_data_kwargs,
     )
 
     if Version(langchain.__version__) >= Version("0.0.311"):
         checker_model = lc_model
         if isinstance(lc_model, str):
+            # TODO: use model_config instead of code_paths[0]
             checker_model = (
                 _load_model_code_path(lc_model, code_paths[0])
```

Comment:
Maybe we can move this code on top when the lc_model is a str, we load the model and make that the lc_model. Lot of things would be solved because of that. What do you think?